snapmodel_policy_test.go

```diff
@@ -0,0 +1,401 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot_test
+
+import (
+	"encoding/base64"
+	"time"
+
+	"github.com/canonical/go-tpm2"
+	. "github.com/snapcore/secboot"
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+
+	. "gopkg.in/check.v1"
+)
+
+type snapModelProfileTest struct{}
+
+var _ = Suite(&snapModelProfileTest{})
+
+type testAddSnapModelProfileData struct {
+	profile *PCRProtectionProfile
+	params  *SnapModelProfileParams
+	values  []tpm2.PCRValues
+}
+
+func (s *snapModelProfileTest) testAddSnapModelProfile(c *C, data *testAddSnapModelProfileData) {
+	profile := data.profile
+	if profile == nil {
+		profile = NewPCRProtectionProfile()
+	}
+	c.Check(AddSnapModelProfile(profile, data.params), IsNil)
+	values, err := profile.ComputePCRValues(nil)
+	c.Assert(err, IsNil)
+	c.Check(values, DeepEquals, data.values)
+	for i, v := range values {
+		c.Logf("Value %d:", i)
+		for alg := range v {
+			for pcr := range v[alg] {
+				c.Logf(" PCR%d,%v: %x", pcr, alg, v[alg][pcr])
+			}
+		}
+	}
+}
+
+func (s *snapModelProfileTest) TestAddSnapModelProfile1(c *C) {
+	s.testAddSnapModelProfile(c, &testAddSnapModelProfileData{
+		params: &SnapModelProfileParams{
+			PCRAlgorithm: tpm2.HashAlgorithmSHA256,
+			PCRIndex:     12,
+			Models: []*asserts.Model{
+				assertstest.FakeAssertion(map[string]interface{}{
+					"type":         "model",
```

Comment:
to be clear these are all UC18 style models,  not UC20 ones,  UC20 ones look like this:

https://github.com/snapcore/models/blob/master/ubuntu-core-20-amd64.model
https://github.com/snapcore/snapd/blob/master/seed/seed20_test.go#L106

Reply:
I've updated the tests to use UC20 style models now.